VerifyEncoding/Test-Encoding.ps1

```diff
@@ -55,7 +55,17 @@ function Test-Encoding
         {
             throw "Cannot call `"git ls-tree`": exit code $LASTEXITCODE."
         }
-        Write-Output "Total files in the repository: $( $allFiles.Length )"
+
+        $submodulesPath = "$SourceRoot/.gitmodules"
+
+        if (Test-Path $submodulesPath) {
+            Write-Host "Filtering submodules ..."
+            $submodules = (Get-Content $submodulesPath -Raw).Trim()
+            $allFiles = $allFiles | Where-Object { $submodules -notmatch [Regex]::Escape($_) }
```

Comment:
Uhh, I think we can do better here :)

To do this one correctly, we'd need to read the `.gitmodules` file and filter the actual file paths by it.

But maybe let's do something simpler.

Is my understanding correct that a submodule is normally a directory, and the special thing about it is the fact that we are getting _a directory_ from the output of `git ls-tree`?

If so, then maybe let's just skip all directories in the loop below? What do you think?

Reply:
Here we can implement this kind of path check

```pwsh
$path = "C:\MyFolder"

if ((Test-Path $path) -and ((Get-Item $path).PSIsContainer)) {
    Write-Host "It's a directory."
} else {
    Write-Host "It's not a directory."
}
```

Reply:
Yep, that works for me.